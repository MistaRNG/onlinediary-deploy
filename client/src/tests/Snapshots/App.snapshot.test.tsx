import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { test, expect } from '@jest/globals';
import App from '../../app/App';

const mockStore = configureStore([]);
const initialState = {
  safety: { locked: false, alarm: null },
  user: { hasCheckedUsername: true, username: null },
  mode: { darkMode: false },
  error: { error: null, trackErrorChange: false },
};

const store = mockStore(initialState);

test('renders App correctly', () => {
  // ARRANGE
  const component = (
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  );

  // ACT
  const tree = renderer.create(component).toJSON();

  // ASSERT
  expect(tree).toMatchSnapshot();
});
