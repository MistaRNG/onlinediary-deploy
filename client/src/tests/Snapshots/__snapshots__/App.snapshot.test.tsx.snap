// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="bp4-dark dark"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseMove={[Function]}
>
  <div
    className="DarkModeButton"
  >
    <button
      className="bp4-button"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      type="button"
    >
      <span
        aria-hidden={true}
        className="bp4-icon bp4-icon-flash"
        icon="flash"
      >
        <svg
          data-icon="flash"
          height={16}
          role="img"
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M4 8c0-.55-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1h2c.55 0 1-.45 1-1zm4-4c.55 0 1-.45 1-1V1c0-.55-.45-1-1-1S7 .45 7 1v2c0 .55.45 1 1 1zM3.79 5.21a1.003 1.003 0 001.42-1.42l-1.5-1.5a1.003 1.003 0 00-1.42 1.42l1.5 1.5zm.71 5.29c-.28 0-.53.11-.71.29l-1.5 1.5a1.003 1.003 0 001.42 1.42l1.5-1.5a1.003 1.003 0 00-.71-1.71zm7-5c.28 0 .53-.11.71-.29l1.5-1.5a1.003 1.003 0 00-1.42-1.42l-1.5 1.5a1.003 1.003 0 00.71 1.71zm.71 5.29a1.003 1.003 0 00-1.42 1.42l1.5 1.5a1.003 1.003 0 001.42-1.42l-1.5-1.5zM15 7h-2c-.55 0-1 .45-1 1s.45 1 1 1h2c.55 0 1-.45 1-1s-.45-1-1-1zM8 5C6.34 5 5 6.34 5 8s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3zm0 4c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm0 3c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    aria-valuemax={100}
    aria-valuemin={0}
    className="bp4-spinner"
    role="progressbar"
  >
    <div
      className="bp4-spinner-animation"
    >
      <svg
        height={50}
        strokeWidth="8.00"
        viewBox="1.00 1.00 98.00 98.00"
        width={50}
      >
        <path
          className="bp4-spinner-track"
          d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
        />
        <path
          className="bp4-spinner-head"
          d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
          pathLength={280}
          strokeDasharray="280 280"
          strokeDashoffset={210}
        />
      </svg>
    </div>
  </div>
</div>
`;
